### PR TITLE
Add X509_NAME_add_entry binding

### PR DESCRIPTION
**Pull request by steffen-eiden**
Signed-off-by: Steffen Eiden <seiden@linux.ibm.com>

### CODE REVIEW

File: openssl-sys/src/handwritten/x509.rs [hunk 1]
```diff
@@ -286,6 +286,13 @@ const_ptr_api! {
         pub fn X509_NAME_dup(x: #[const_ptr_if(ossl300)] X509_NAME) -> *mut X509_NAME;
         #[cfg(any(ossl110, libressl270))]
         pub fn X509_dup(x: #[const_ptr_if(ossl300)] X509) -> *mut X509;
+
+        pub fn X509_NAME_add_entry(
+            name: *mut X509_NAME,
+            ne: #[const_ptr_if(any(ossl110, libressl))] X509_NAME_ENTRY,
```
Comment (sfackler) [inline openssl-sys/src/handwritten/x509.rs:292]: It looks like this may be incorrect for old libressl versions.

**Comment by sfackler** [comment]
LGTM other than the test failures.

**Comment by steffen-eiden** [comment]
Tests succeed now, changed the required libressl version to 3.5.x+

**Comment by sfackler** [comment]
Thanks!

**Pull request merged**